docs/contributing-guide.md

```diff
@@ -0,0 +1,168 @@
+1. `config` contains Scala 2.10 and 2.12 module to read and write configuration files.
+1. `docs` and `docs-gen` define our docs infrastructure.
+1. `benchmark-bridge` and `benchmarks` define our compiler benchmark infrastructure.
+1. `launcher` is an application to spawn a bloop server and establish a
+bsp/cli connection to it.
+1. `nailgun` is a git submodule of `scalacenter/nailgun`, a fork of `facebook/nailgun`.
+1. `sockets` is a library to use Unix Domain sockets and Named Pipes in Windows.
+1. `website` contains the [Docusaurus](https://docusaurus.io/) code of our website. 
+1. `zinc` is a git submodule to the version of Zinc that bloop master is using.
+
+## Set the repository up
+
+1. Clone the [GitHub repository](https://github.com/scalacenter/bloop)
+1. Change directory to bloop's base directory and run `git submodule update --init`.
+1. Run the sbt shell in the base directory.
+
+If the sbt shell did not fail, you already have a working bloop build in your
+machine.
+
+To hack on the bloop repository you can use either
+[Metals](https://scalameta.org/metals/) or
+[IntelliJ](https://www.jetbrains.com/idea/). Open the project and import the
+build in order to get started.
+
+## Learn the developer workflow
+
+The recommended way to build and hack on bloop is by using bloop. Run
+`bloopInstall` if you are not using Metals to export the build to the
+`.bloop/` configuration directory.
+
+### Common commands
+
+Once the build is exported, here are a few commands to test your changes.
+
+1. `bloop test backend` runs unit tests for the `backend/` project.
+1. `bloop test frontend` runs unit tests for most of the build server. This is the most common command to run.
+1. `bloop test frontend -o bloop.CompileSpec` will test only the compilation spec.
+1. `bloop test frontend -o bloop.DeduplicationSpec -- "bloop.DeduplicationSpec.three concurrent clients deduplicate compilation"` will test only that particular test case in the `DeduplicationSpec` test suite.
+
+### Write commit messages
+
+Write concise commit messages, with a capital letter at the beginning of the
+commit message and with a potential description of what your change does so
+that future code wanderers can understand what your contribution is.
+
+**Tip**: If you're only making changes to the docs, add `[DOCS]` to the commit
+message so that Bloop does not run tests for your commit.
+
+## Install a SNAPSHOT release
+
+When a PR is merged in `scalacenter/bloop`, the CI cuts a new release
+available to any Bloop user. The version number of the latest SNAPSHOT
+release can be found in the version table available in the [Installation
+guide](http://localhost:3000/bloop/setup). It has a stable version number to
+make dependencies reproducible.
+
+### Depend on SNAPSHOT releases
+
+If you are using bloop as a library dependency, you can add Scala Center
+Bintray's resolver and use the version right away.
```

Comment:
Would be nice to either give the steps for that, or point to somewhere that does